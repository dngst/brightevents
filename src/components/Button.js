import styled from "styled-components";

const Button = styled.button`
  color: ${(props) => (props.primary ? "#ffffff" : "#f78f1e")};
  background-color: ${(props) => (props.primary ? "#f78f1e" : "#ffffff")};
  border: 1px solid #f78f1e;
  border-radius: 0;
  margin: auto;
  padding: 0.7em 2em;
  font-size: 0.9em;
  text-transform: uppercase;
  font-weight: bold;
  cursor: pointer;
  width: ${(props) => (props.max ? "max-content" : "100%")};
`;

export default Button;
