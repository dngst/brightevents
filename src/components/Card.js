import styled from "styled-components";

const Card = styled.div`
  height: 100%;
  margin: 0 0 4%;
  border: 2px dashed ${(props) => props.theme.blue};
  text-align: left;
  padding: 0 4%;
  p {
    font-size: 1rem;
    line-height: 1.2;
  }
  .event-title {
    font-size: 1.2rem;
    text-transform: uppercase;
    font-family: sans-serif;
    a {
      color: ${(props) => props.theme.grey};
      text-decoration: underline;
    }
    span {
      color: ${(props) => props.theme.highlight};
      font-weight: bold;
    }
  }
  .event-cat {
    font-size: 0.9rem;
  }
  @media (max-width: 480px) {
    padding: 0;
    border: none;
    .event-title {
      span {
        display: none;
      }
    }
  }
`;

export default Card;
