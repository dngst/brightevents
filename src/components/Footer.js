import styled from "styled-components";

const AppFooter = styled.footer`
  display: flex;
  justify-content: space-evenly;
  font-size: 0.7rem;
  background-color: ${(props) => props.theme.grey};
  position: absolute;
  bottom: 0;
  width: 100%;
  @media (max-width: 768px) {
    justify-content: space-between;
    display: block;
  }
`;

const AppName = styled.p`
  color: ${(props) => props.theme.black};
  @media (max-width: 768px) {
    margin: 2% 2%;
  }
`;

const Footer = () => {
  return (
    <AppFooter>
      <AppName>twitter.</AppName>
      <AppName>instagram.</AppName>
      <AppName>bizz stuff.</AppName>
      <AppName>&copy; 2022 Bright Events</AppName>
    </AppFooter>
  );
};

export default Footer;