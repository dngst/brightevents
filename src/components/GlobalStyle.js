import { createGlobalStyle } from "styled-components";

const GlobalStyle = createGlobalStyle`
  body {
  	font-family: 'Quicksand', sans-serif;
    letter-spacing: 0.0625em;
    display:flex;
    flex-direction:column;
    margin: 0;
    transition: all 0.4s ease-in 0s;
  }

  a {
    text-decoration: none;
    color: ${(props) => props.theme.link};
  }

  body::-webkit-scrollbar {
    width: 0.35em;
  }

  body::-webkit-scrollbar-thumb {
    background-color: ${(props) => props.theme.link};
    &:hover {
      background-color: ${(props) => props.theme.highlight};
    }
  }

  .event-row {
    display: flex;
    justify-content: space-between;
    p {
      text-transform: uppercase;
      font-size: 1.05rem;
      &:nth-child(1) {
        font-weight: 700;
        color: ${(props) => props.theme.highlight};
      }
      &:nth-child(2) {
        font-weight: 900;
      }
      &:nth-child(3) {
        span {
          font-weight: 700;
        }
      }
      &:nth-child(4) {
        font-weight: 900;
      }
    }

    @media (max-width: 768px) {
      display: block;
      text-align: left;
      margin-bottom: 13%;
    }
  }

  main {
    position: relative;
    min-height: 100vh;
  }

  #root {
    @media (max-width: 768px) {
      margin-bottom: 10%;
    }
  }
`;

export default GlobalStyle;
