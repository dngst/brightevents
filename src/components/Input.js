import styled from "styled-components";

export const Input = styled.input.attrs((props) => ({
  type: "text",
}))`
  border: none;
  border-bottom: 1px solid
    ${(props) => (props.transparent ? props.theme.white : props.theme.brown)};
  display: block;
  height: 2rem;
  width: 100%;
  margin: 0 0 1em;
  box-sizing: border-box;
  ::placeholder {
    color: ${(props) => props.theme.brown};
  }
  &:focus {
    outline: none;
    border-color: ${(props) => props.theme.white};
  }
`;

export const PasswordInput = styled(Input).attrs({
  type: "password",
})``;

export const TextArea = styled.textarea`
  border: none;
  border-bottom: 1px solid ${(props) => props.theme.brown};
  height: 6rem;
  width: 100%;
  margin: 0 0 1em;
  box-sizing: border-box;
  overflow: auto;
  resize: none;
  ::placeholder {
    color: ${(props) => props.theme.brown};
  }
  &:focus {
    outline: none;
    border-color: ${(props) => props.theme.white};
  }
`;

export const TimePicker = styled(Input).attrs({
  type: "time",
})``;

export const DatePicker = styled(Input).attrs({
  type: "date",
})``;

export default Input;
