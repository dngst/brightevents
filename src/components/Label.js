import styled from "styled-components";

export const Label = styled.label`
  align-items: left;
  display: flex;
  font-size: 0.8em;
`;

export const LabelText = styled.span``;
