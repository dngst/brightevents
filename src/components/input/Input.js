import styled from "styled-components";

const Input = styled.input.attrs(props => ({
  type: 'text'
}))`
  border: 1px solid #f78f1e;
  display: block;
  height: 2rem;
  width: 100%;
  margin: 0 0 1em;
  text-transform: capitalize;
  border-radius: 0;
  box-sizing: border-box;
  ::placeholder {
    color: #f78f1e;
  }
  &:focus {
    outline: none;
    border-color: #ffffff;
  }
`

export default Input;
