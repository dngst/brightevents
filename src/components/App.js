import { Route, Switch } from "wouter";
import GlobalStyle from "components/GlobalStyle";
import Navbar from "components/nav/Navbar";
import Home from "pages/Home";
import CreateEvent from "pages/CreateEvent";
import Login from "pages/Login";
import SignUp from "pages/SignUp";
import ResetPassword from "pages/ResetPassword";
import ViewEvents from "pages/ViewEvents";
import Event from "pages/Event";
import UpdateEvent from "pages/UpdateEvent";
import PageNotFound from "pages/PageNotFound";
import About from "pages/About";

const App = () => {
  return (
    <div>
      <GlobalStyle />
      <Navbar />
      <Switch>
        <Route path="/" component={Home} />
        <Route path="/event" component={Event} />
        <Route path="/events" component={ViewEvents} />
        <Route path="/new" component={CreateEvent} />
        <Route path="/update" component={UpdateEvent} />
        <Route path="/signup" component={SignUp} />
        <Route path="/login" component={Login} />
        <Route path="/resetpassword" component={ResetPassword} />
        <Route path="/about" component={About} />
        <Route component={PageNotFound} />
      </Switch>
    </div>
  );
};

export default App;
