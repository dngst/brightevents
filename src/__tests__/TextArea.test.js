import { render } from "@testing-library/react";
import { TextArea } from "components/Input";

it("works", () => {
  const { container } = render(<TextArea />);
  expect(container.firstChild).toMatchSnapshot();
});

it("applies default styles", () => {
  const { container } = render(<TextArea />);
  expect(container.firstChild).toHaveStyleRule("border", "none");
  expect(container.firstChild).toHaveStyleRule("text-transform", "capitalize");
  expect(container.firstChild).toHaveStyleRule("overflow", "auto");
  expect(container.firstChild).toHaveStyleRule("resize", "none");
  expect(container.firstChild).toHaveStyleRule("outline", "none", {
    modifier: "&:focus",
  });
});
