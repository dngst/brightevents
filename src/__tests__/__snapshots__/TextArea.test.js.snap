// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
.c0 {
  border: none;
  border-bottom: 1px solid;
  height: 6rem;
  width: 100%;
  margin: 0 0 1em;
  text-transform: capitalize;
  box-sizing: border-box;
  overflow: auto;
  resize: none;
}

.c0:focus {
  outline: none;
}

.c0:focus {
  outline: none;
}

.c0:focus {
  outline: none;
}

.c0:focus {
  outline: none;
}

<textarea
  class="c0"
/>
`;
